chainservice/builder.go

```diff
@@ -298,6 +298,9 @@ func (builder *Builder) buildSGDRegistry(forTest bool) error {
 }
 
 func (builder *Builder) buildContractStakingIndexer(forTest bool) error {
+	if !builder.cfg.Chain.EnableStakingProtocol {
```

Comment:
will `contractStakingIndexer` be `nil` when enabled?

Reply:
it could be, if `cfg.Genesis.SystemStakingContractAddress == ""` in the config

Reply:
should ContractStakingIndexer be built when cfg.Genesis.SystemStakingContractAddress == ""

Reply:
no, in that case, means SystemStaking (IIP-13) is not enabled on the chain, no need to build the indexer